src/mlpack/core/util/deprecated.hpp

```diff
@@ -0,0 +1,19 @@
+/**
+ * @file deprecated.hpp
+ * @author Marcos Pividori.
+ *
+ * Definition of the DEPRECATED macro.
+ */
+#ifndef MLPACK_CORE_UTIL_DEPRECATED_HPP
+#define MLPACK_CORE_UTIL_DEPRECATED_HPP
+
+#ifdef __GNUG__
+#define DEPRECATED __attribute__((deprecated))
+#elif defined(_MSC_VER)
+#define DEPRECATED __declspec(deprecated)
+#else
+#pragma message("WARNING: You need to implement DEPRECATED for this compiler")
+#define DEPRECATED
```

Comment:
Do you think we could call this `mlpack_deprecated` so it has the same format (lowercase) as the few other mlpack macros?